package/MDAnalysis/core/topologyattrs.py

```diff
@@ -291,6 +298,12 @@ def __init__(cls, name, bases, classdict):
         if attrname is None:
             attrname = singular
 
+        # add intra connections
```

Comment:
Maybe it would be cleaner to add the transplant in the `mda.core.topologyattr._Connection` class rather than here?

Reply:
Yeah I'll add them manually, it's a bit repetitive but is much less likely to go catastrophically wrong.

Reply:
If you add it to the base class, you should be able to add it only once.

Reply:
They each have different names, I'm not sure how to distinguish between `intra_bonds`, `intra_angles` etc. in the base class :/

Reply:
This wound up being more of a pain than I expected (the below would need to be manually added for each class, unless I'm overlooking something obvious):

```python
	def intra_bonds(self):
        """Get bonds only within this AtomGroup
        """
        return self.get_connections("bonds", outside=False)

    transplants[AtomGroup].append(
        ('intra_bonds', property(intra_bonds, None, None,
                                 intra_bonds.__doc__))
    )
```

Instead I could tighten up the base-class checking to only work for `_Connection` subclasses (forgot the underscore before). In your opinion is that safe enough?

Reply:
Hum... I had the level of magic wrong when doing my suggestion. I do not know if it is possible to attach a metaclass to `_Connection` knowing that it inherits from `AtomAttr` which itself inherit from `TopologyAttr` that has a metaclass already. If it is possible, then it would be a good solution to keep the responsibilities tidy. If not, then I live with your current solution: only so much effort is worth spending on my pernickettyness.

Reply:
🤦‍♀️ Yes that's a much better solution. Luckily, everything in Python is mutable.